common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -140,18 +141,52 @@ public Object run() {
+            // https://openjdk.org/jeps/471
             if (unsafe != null) {
                 final Unsafe finalUnsafe = unsafe;
                 final Object maybeException = AccessController.doPrivileged(new PrivilegedAction<Object>() {
                     @Override
                     public Object run() {
                         try {
-                            finalUnsafe.getClass().getDeclaredMethod(
+                            // Other methods like storeFence() and invokeCleaner() are tested for elsewhere.
+                            Class<? extends Unsafe> cls = finalUnsafe.getClass();
+                            cls.getDeclaredMethod("objectFieldOffset", Field.class);
+                            cls.getDeclaredMethod("staticFieldOffset", Field.class);
+                            cls.getDeclaredMethod("staticFieldBase", Field.class);
+                            cls.getDeclaredMethod("arrayBaseOffset", Class.class);
+                            cls.getDeclaredMethod("arrayIndexScale", Class.class);
+                            cls.getDeclaredMethod("allocateMemory", long.class);
+                            cls.getDeclaredMethod("reallocateMemory", long.class, long.class);
+                            cls.getDeclaredMethod("freeMemory", long.class);
+                            cls.getDeclaredMethod("setMemory", long.class, long.class, byte.class);
+                            cls.getDeclaredMethod("setMemory", Object.class, long.class, long.class, byte.class);
+                            cls.getDeclaredMethod(
                                     "copyMemory", Object.class, long.class, Object.class, long.class, long.class);
+                            cls.getDeclaredMethod("getBoolean", Object.class, long.class);
+                            cls.getDeclaredMethod("getByte", long.class);
+                            cls.getDeclaredMethod("getByte", Object.class, long.class);
+                            cls.getDeclaredMethod("getInt", long.class);
+                            cls.getDeclaredMethod("getInt", Object.class, long.class);
+                            cls.getDeclaredMethod("getLong", long.class);
+                            cls.getDeclaredMethod("getLong", Object.class, long.class);
+                            cls.getDeclaredMethod("putByte", long.class, byte.class);
+                            cls.getDeclaredMethod("putByte", Object.class, long.class, byte.class);
+                            cls.getDeclaredMethod("putInt", long.class, int.class);
+                            cls.getDeclaredMethod("putInt", Object.class, long.class, int.class);
+                            cls.getDeclaredMethod("putLong", long.class, long.class);
+                            cls.getDeclaredMethod("putLong", Object.class, long.class, long.class);
+                            cls.getDeclaredMethod("addressSize");
+                            // The following may throw UnsupportedOperationException:
```

Comment:
Maybe this could be rephrased to something like "The following tests that we can allocate raw memory and if not we'll expect it to throw an UnsupportedOperationException".